lib/AST/ModuleDependencies.cpp

```diff
@@ -516,19 +525,9 @@ swift::dependencies::registerCxxInteropLibraries(
                       return mainModuleName == Name;
                     })) {
     // Only link with CxxStdlib on platforms where the overlay is available.
-    switch (Target.getOS()) {
-    case llvm::Triple::Win32: {
-      RegistrationCallback(
-          LinkLibrary(hasStaticCxxStdlib ? "libswiftCxxStdlib" : "swiftCxxStdlib",
-                      LibraryKind::Library));
-      break;
-    }
-    default:
-      if (Target.isOSDarwin() || Target.isOSLinux())
-        RegistrationCallback(LinkLibrary("swiftCxxStdlib",
-                                         LibraryKind::Library));
-      break;
-    }
+    if (Target.isOSDarwin() || Target.isOSLinux() || Target.isOSWindows())
+      RegistrationCallback(LinkLibrary{"swiftCxxStdlib", LibraryKind::Library,
+                                       hasStaticCxxStdlib});
```

Comment:
This is missing a parameter.

Reply:
The fourth parameter is optional, defaults to `false`.

Reply:
I see. I thought this parameter is for `forceLoad` and was wrongly used for static. Now I see what you mean by using static to select if the prefix has `lib` or not.

Will be good to have test for the change.

Reply:
This is tested now - we had special cased the C++ interop libraries (which have test coverage) and will now be through this path. If there something else that you have in mind?